File: byteback-test/test-java-17/src/main/java/byteback/test/patternmatching/Basic.java
/**
 * RUN: %{byteback} -cp %{jar} -c byteback.test.patternmatching.Basic -o %t.mlw
 */
package byteback.test.patternmatching;

public class Basic {

	public static int test() {
		Object obj = new int[10];

		if (obj instanceof int[] matched) {
			int len = matched.length;

			return len;
		}

		return 0;
	}

}
/**
 * RUN-IGNORE: %{verify} %t.bpl | filecheck %s
 * CHECK-IGNORE: Boogie program verifier finished with 2 verified, 0 errors
 */

File: byteback-test/test-kotlin-1.8.0/src/main/kotlin/byteback/test/controlflow/Basic.kt
/**
 * RUN: %{translate} %{byteback} %{jar} %s %t
 */

package byteback.test.controlflow

class Basic {

		fun sum(a: Int, b: Int): Int {
				return a + b
		}

		fun empty(): Unit {
		}

		fun doubleAssignment(): Unit {
				var a = 0
				a = a + 42
		}

		fun emptyWhile(a: Boolean): Unit {
				while (a) {
				}
		}

		fun emptyDoWhile(a: Boolean): Unit {
				do {
				} while (a);
		}

		fun emptyIf(a: Boolean): Unit {
				if (a) {
				}
		}

}

/**
 * RUN: %{verify} %t
 * CHECK-IGNORE: Boogie program verifier finished with 7 verified, 0 errors
 */
